lib/bunyan.js

```diff
@@ -1184,6 +1168,52 @@ function safeCycles() {
     };
 }
 
+// A JSON stringifier that handles cycles safely - tracks seen values in a Set.
+function safeCyclesSet() {
+    var seen = new Set();
+    return function (key, val) {
+        if (!val || typeof (val) !== 'object') {
+            return val;
+        }
+        if (seen.has(val)) {
+            return '[Circular]';
+        }
+        seen.add(val);
+        return val;
+    };
+}
+
+// A JSON stringifier that handles cycles safely.
+// Usage: JSON.stringify(obj, safeCycles())
+//
+// Choose the best safe cycle function from what is available - issue#445.
+var safeCycles = Set ? safeCyclesSet : safeCyclesArray;
+
+// A safer JSON.stringify that handles cycles and getter exceptions (when
+// safeJsonStringify is installed).
+function safeStringify(rec) {
+    try {
+        return JSON.stringify(rec);
```

Comment:
Comment here that first attempt is for raw speed, and that an exception is thrown for a cycle.